File: skills/tests/resources/IntegerMultiplyBy2RapidsUDF.java
package com.udf;

import org.apache.spark.sql.api.java.UDF1;
import ai.rapids.cudf.*;
import com.nvidia.spark.RapidsUDF;

public class IntegerMultiplyBy2RapidsUDF implements UDF1<Integer, Integer>, RapidsUDF {
  @Override
  public Integer call(Integer value) {
    return value == null ? null : value * 2;
  }

  @Override
  public ColumnVector evaluateColumnar(int numRows, ColumnVector... args) {
    try (Scalar two = Scalar.fromInt(2)) {
      return args[0].mul(two);
    }
  }
}

File: skills/tests/resources/IntegerMultiplyBy2RapidsUDF.scala
/*
 * SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

package com.udf

import ai.rapids.cudf._
import com.nvidia.spark.RapidsUDF
import Arm.withResource

class IntegerMultiplyBy2RapidsUDF extends Function1[Integer, Integer] with Serializable with RapidsUDF {
  override def apply(value: Integer): Integer = {
    if (value == null) null else value * 2
  }

  override def evaluateColumnar(numRows: Int, args: ColumnVector*): ColumnVector = {
    withResource(Scalar.fromInt(2)) { two =>
      args.head.mul(two)
    }
  }
}
